lib/pbench/server/filetree.py

```diff
@@ -336,72 +336,72 @@ def unpack(self, incoming: Path, results: Path):
+                    "--file=" + str(self.tarball_path),
                 ],
                 cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballUnpackError(self.name, str(exc)) from exc
+            raise TarballUnpackError(self.tarball_path, str(exc)) from exc
         else:
             if process.returncode != 0:
                 raise TarballUnpackError(
-                    self.name,
+                    self.tarball_path,
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
+        find_command = "find . ( -type d -exec chmod ugo+rx '{}' + ) -o ( -type f -exec chmod ugo+r '{}' + )"
         try:
             process = subprocess.run(
-                command.split(),
+                find_command.split(),
                 cwd=unpacked,
                 stdin=subprocess.DEVNULL,
                 capture_output=True,
                 text=True,
             )
         except Exception as exc:
-            raise TarballModeChangeError(self.name, str(exc)) from exc
+            raise TarballModeChangeError(self.tarball_path, str(exc)) from exc
         else:
             if process.returncode != 0:
                 raise TarballModeChangeError(
-                    self.name,
+                    self.tarball_path,
                     f"tar exited with status {process.returncode}:  {process.stderr.strip()!r}",
                 )
 
-        self.unpacked = incoming
+        self.unpacked = incoming / self.name
         unpack = unpacked / self.name
 
         try:
-            shutil.move(str(unpack), str(self.unpacked))
+            shutil.move(unpack, incoming)
         except Exception as exc:
-            raise TarballUnpackError(
-                self.name,
-                f"Error in moving unpacked tarball to Unpacked directory: {str(exc)}",
-            )
             try:
-                os.remove(self.unpacked / self.name)
+                shutil.rmtree(self.unpacked)
             except FileNotFoundError:
                 raise TarballUnpackError(
-                    self.name, "Error in removing partially copied tarball"
+                    self.tarball_path, "Error in removing partially copied Tarball"
                 )
+            raise TarballUnpackError(
+                self.tarball_path,
+                f"Error in moving unpacked tarball to Unpacked directory: {str(exc)}",
+            )
 
         try:
-            os.remove(unpacked)
+            os.rmdir(unpacked)
         except Exception as exc:
             raise TarballUnpackError(
-                self.name,
+                self.tarball_path,
                 f"Removal of unpack directory failed: '{str(exc)}'",
             )
```

Comment:
I'm confused.  I think the steps here should be:
1. Unpack (`tar`) the tarball into a temporary directory
2. Fix the file permissions (`find`)
3. Move the temporary directory (`move()`) to its final location

The rest of this code is error handling:
1. If the `tar` command or its invocation fails, raise an exception
2. If the `find` command or its invocation fails, raise an exception
3. If the directory-move (`move()`) fails, remove the directory (`rmtree()`) and raise an exception.
3a. If the directory-removal (`rmtree()`) fails, raise a different exception.

I think you've got all this covered.  So, the question is, what does this `os.rmdir()` call do?  At this point, everything should be done...I think...so there shouldn't be anything left to remove...is that right?  (Or, did we create a "working directory" that this is intended to remove?)

Reply:
`Shutil.rmtree()` is added if the tarball move is partially successful. So if at the time of moving we face any error those will be handled by `shutil.rmtree()`. `shutil.rmtree()` is removing the partialy copied tarball in incoming tree.
`os.rmdir()` is working after move has worked successfully, so we are removing an empty `${INCOMING}/{controller}/{name}.unpack` directory.